### PR TITLE
doc/user: add guide for Power BI

### LINKED ISSUE #21652: doc/user: add guide for Power BI
### Documentation request

Connecting to Power BI is supported in Materialize as of [v0.68](https://materialize.com/docs/releases/v0.68/).

See [this thread](https://materializeinc.slack.com/archives/C04SN4BEQ0N/p1694097773375529) for more details.

### LINKED ISSUE #21777: Document that PowerBI doesn't list materialized views and workarounds.
### Feature request

slack thread: https://materializeinc.slack.com/archives/CU7ELJ6E9/p1694798226507819?thread_ts=1694787975.181369&cid=CU7ELJ6E9

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

Adding a Power BI guide.

Closes #21652
Closes #21777

### CODE REVIEW

File: doc/user/content/serve-results/power-bi.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+## Database connection details
+
+To set up a connection from Power BI to Materialize, use the native
+[PostgreSQL database driver](https://learn.microsoft.com/en-us/power-query/connectors/postgresql#connect-to-a-postgresql-database-from-power-query-desktop)
+with the following parameters:
+
+Field                  | Value
+---------------------- | ----------------
+Database type          | **PostgreSQL database**
+Server                 | Materialize host name followed by `:6875`
```
Comment (sthm) [inline doc/user/content/serve-results/power-bi.md:26]: Does it make sense to add a concrete example here, like `id.us-east-1.aws.materialize.cloud:6875`?

File: doc/user/content/serve-results/power-bi.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+Database               | **materialize**
+Data Connectivity mode | **DirectQuery**
+Database username      | Materialize user.
+Database password      | App-specific password.
```
Comment (sthm) [inline doc/user/content/serve-results/power-bi.md:30]: Should we point users to the docs that explain how to generate an app password?

**Review by sthm — APPROVED** [review]
Looks good, thanks!

File: doc/user/content/serve-results/power-bi.md (hunk 2 above)
Comment (bobbyiliev) [reply doc/user/content/serve-results/power-bi.md:30]: I've followed the structure for the other BI guides that we have like this one here:
> https://materialize.com/docs/serve-results/deepnote/#step-1-create-an-integration

Also, there isn't a dedicated guide on how to generate an app password I believe.

File: doc/user/content/serve-results/power-bi.md (hunk 1 above)
Comment (bobbyiliev) [reply doc/user/content/serve-results/power-bi.md]: Added this as an example

File: doc/user/content/serve-results/power-bi.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    weight: 10
+---
+
+You can use [Power BI](https://powerbi.microsoft.com/) to create dashboards
+based on the data maintained in Materialize.
+
+## Database connection details
+
+To set up a connection from Power BI to Materialize, use the native
+[PostgreSQL database driver](https://learn.microsoft.com/en-us/power-query/connectors/postgresql#connect-to-a-postgresql-database-from-power-query-desktop)
```
Comment (morsapaes) [inline doc/user/content/serve-results/power-bi.md:20]: Are there no considerations wrt supported driver and/or Power BI versions?

**Review by morsapaes — APPROVED** [review]

**Comment by morsapaes** [comment]
Thanks, @bobbyiliev! Could you just edit the Power BI entry under `/integrations` to point to the guide?

Also: I'd like us to start adding a bit more meat to these guides, with details like [how to setup access for multiple users](https://materializeinc.slack.com/archives/C04SN4BEQ0N/p1695137015906799), dos and don'ts. No need to block the PR on that, though!

**Comment by bobbyiliev** [comment]
> Thanks, @bobbyiliev! Could you just edit the Power BI entry under `/integrations` to point to the guide?
> 
> Also: I'd like us to start adding a bit more meat to these guides, with details like [how to setup access for multiple users](https://materializeinc.slack.com/archives/C04SN4BEQ0N/p1695137015906799), dos and don'ts. No need to block the PR on that, though!

Thank you @morsapaes! I've just added that reference to the `/integrations` page as well.

Also extended the guide to include the known Power BI limitation around materialized views as requested in https://github.com/MaterializeInc/materialize/issues/21777

**Review by sjwiesman — APPROVED** [review]

**Pull request merged**